scripts/ci/forbidden-identifiers.js

```diff
@@ -34,68 +28,135 @@ const blocked_statements = [
-const componentFolders = fs
-  .readdirSync(`${__dirname}/../../src/components`)
-  .map(componentName => `src/components/${componentName}`);
-
-const scopePackages = ['src/core'].concat(componentFolders);
-
+const sourceFolders = ['./src', './e2e'];
+const scopePackages = ['src/core'].concat(glob('src/components/*'));
 const blockedRegex = new RegExp(blocked_statements.join('|'), 'g');
 const importRegex = /from\s+'(.*)';/g;
 
-/**
- * Find the fork point between HEAD of the current branch, and master.
- * @return {Promise<string>} A promise which resolves with the fork SHA (or reject).
+/*
+ * Verify that the current PR is not adding any forbidden identifiers.
+ * Run the forbidden identifiers check against all sources when not verifying a PR.
  */
-function findForkPoint() {
-  return exec('git merge-base --fork-point HEAD master')
-    .then(function(stdout) {
-      return stdout.split('\n')[0];
+
+resolveFiles()
+
+  /* Only match .js or .ts, and remove .d.ts files. */
+  .then(fileList => {
+    return fileList.filter(name => name.match(/\.(js|ts)$/) && !name.match(/\.d\.ts$/));
+  })
+
+  /* Read content of the filtered files */
+  .then(fileList => {
+    return fileList.map(fileName => {
+      return {
+        name: fileName,
+        content: fs.readFileSync(fileName, 'utf-8')
+      };
     });
-}
+  })
 
-/**
- * Get the commit range to evaluate when this script is run.
- * @return {Promise<string>} A commit range of the form ref1...ref2.
- */
-function getCommitRange() {
-  if (process.env['TRAVIS_COMMIT_RANGE']) {
-    return Promise.resolve(process.env['TRAVIS_COMMIT_RANGE']);
-  } else {
-    return findForkPoint().then((forkPointSha) => `${forkPointSha}...HEAD`);
-  }
-}
+  /* Run checks against content of the filtered files. */
+  .then(diffList => {
+
+    return diffList.reduce((errors, diffFile) => {
+      let fileName = diffFile.name;
+      let content = diffFile.content.split('\n');
+      let lineCount = 0;
+
+      content.forEach(line => {
+        lineCount++;
+
+        let matches = line.match(blockedRegex);
+        let scopeImport = isRelativeScopeImport(fileName, line);
+
+        if (matches || scopeImport) {
+
+          let error = {
+            fileName: fileName,
+            lineNumber: lineCount,
+            statement: matches && matches[0] || scopeImport.invalidStatement
+          };
+
+          if (scopeImport) {
+            error.messages = [
+              'You are using an import statement, which imports a file from another scope package.',
+              `Please import the file by using the following path: ${scopeImport.scopeImportPath}`
+            ];
+          }
+
+          errors.push(error);
+        }
+      });
+
+      return errors;
+
+    }, []);
+  })
+
+  /* Print the resolved errors to the console */
+  .then(errors => {
+    if (errors.length > 0) {
+      console.error('Error: You are using a statement in your commit, which is not allowed.\n');
```

Comment:
While you're in here, can you remove the common in this line? Or we could also change it to something like 
`"You are using one or more blocked statements:"`